src/index.js

```diff
@@ -4,30 +4,91 @@ import { Router } from 'react-router-dom';
 import { ApolloProvider } from 'react-apollo';
 import { ApolloClient } from 'apollo-client';
 import { HttpLink } from 'apollo-link-http';
+import { WebSocketLink } from 'apollo-link-ws';
 import { InMemoryCache } from 'apollo-cache-inmemory';
+import { split, ApolloLink } from 'apollo-link';
+import { getMainDefinition } from 'apollo-utilities';
+import { withClientState } from 'apollo-link-state';
+import { USER_DAILY_REPORTS_COUNT_QUERY } from 'containers/DailyReportPage/DailyReportContainer';
+
 import App from 'containers/App';
 import history from './utils/history';
 
 import 'semantic-ui-css/semantic.min.css';
 import './ress.min.css';
 import './common.style';
 
-const uri = process.env.NODE_ENV === 'production'
-  ? 'Set uri for production here'
-  : process.env.REACT_APP_GRAPHQL_URI;
+const httpUri =
+  process.env.NODE_ENV === 'production'
+    ? 'Set uri for production here'
+    : process.env.REACT_APP_GRAPHQL_URI;
+
+const httpLink = new HttpLink({
+  credentials: 'include',
+  uri: httpUri
+});
+
+const wsUri =
+  process.env.NODE_ENV === 'production'
+    ? 'Set uri for production here'
+    : process.env.REACT_APP_WS_URI;
+
+const wsLink = new WebSocketLink({
+  uri: wsUri,
+  options: {
+    reconnect: true
+  }
+});
+
+const link = split(
+  ({ query }) => {
+    const { kind, operation } = getMainDefinition(query);
+    return kind === 'OperationDefinition' && operation === 'subscription';
+  },
+  wsLink,
+  httpLink
+);
+
+const defaultState = {
```

Comment:
From line 52 to line 81, I'm initializing `apollo-link-state` based on the document
https://www.apollographql.com/docs/link/links/state.html